### PR TITLE
[stable/sonarqube] Update default service type to ClusterIP

### LINKED ISSUE #17115: [stable/sonarqube] Service creates ELB when deployed to AWS EKS
**Describe the bug**
When deploying *stable/sonarqube* to an AWS EKS cluster an unwanted AWS ELB (load balancer) is being created.

**Version of Helm and Kubernetes**:

```
kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.10-eks-5ac0f1", GitCommit:"5ac0f1d9ab2c254ea2b0ce3534fd72932094c6e1", GitTreeState:"clean", BuildDate:"2019-08-20T22:39:46Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
```

```
helm version
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
```

**Which chart**:
*stable/sonarqube*

**What happened**:
An unwanted AWS ELB is created.

**What you expected to happen**:
No external resources to be created unless specified.

**How to reproduce it** (as minimally and precisely as possible):
Install *stable/sonarqube* onto AWS EKS.

**Anything else we need to know**:
N/A

### PR DESCRIPTION
**Pull request by stevehipwell**
Signed-off-by: Steve Hipwell <steve.hipwell@gmail.com>

#### What this PR does / why we need it:
Changes the default service type from `LoadBalancer` to `ClusterIP` as well as making the port handling between service and deployment easier to understand. Without this change this chart will create external resources by default if installed to a cluster where this is possible. This is a safer default value.

#### Which issue this PR fixes
  - fixes #17115

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x ] Chart Version bumped
- [x ] Variables are documented in the README.md
- [ x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/sonarqube/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: sonarqube
 description: Sonarqube is an open sourced code quality scanning tool
-version: 2.2.0
+version: 2.2.1
```
Comment (rjkernick) [inline stable/sonarqube/Chart.yaml]: This would be a major version bump as this could break people's current config

**Review by rjkernick — CHANGES_REQUESTED** [review]

File: stable/sonarqube/Chart.yaml (hunk 1 above)
Comment (stevehipwell) [reply stable/sonarqube/Chart.yaml]: @rjkernick - Do you have a version preference? Is it worth aligning with the Sonarqube version?

Comment (rjkernick) [reply stable/sonarqube/Chart.yaml]: A bump to 3.0.0 would be fine. Just needs to reflect the level of change to the chart.

Comment (stevehipwell) [reply stable/sonarqube/Chart.yaml]: If we're changing the major version then I'd also suggest changing the default deployment strategy to `Recreate` as Sonarqube is a single node chart which has PVC issues when used with `RollingUpdate`.

Comment (stevehipwell) [reply stable/sonarqube/Chart.yaml]: @rjkernick can we progress this?

**Comment by stevehipwell** [comment]
@rjkernick if you're happy with this could you flag it?

**Pull request merged**